db/00083/UpdateGeolocationCvName.pm

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env perl
+
+
+=head1 NAME
+
+ UpdateGeolocationCvName.pm
+
+=head1 SYNOPSIS
+
+mx-run UpdateGeolocationCvName [options] -H hostname -D dbname -u username [-F]
+
+this is a subclass of L<CXGN::Metadata::Dbpatch>
+see the perldoc of parent class for more details.
+
+=head1 DESCRIPTION
+This patch updates the geolocation property cvname to the chado default of 'geolocations_property'
+This subclass uses L<Moose>. The parent class uses L<MooseX::Runnable>
+
+=head1 AUTHOR
+
+Bryan Ellerbrock
+
+=head1 COPYRIGHT & LICENSE
+
+Copyright 2010 Boyce Thompson Institute for Plant Research
+
+This program is free software; you can redistribute it and/or modify
+it under the same terms as Perl itself.
+
+=cut
+
+
+package UpdateGeolocationCvName;
+
+use Moose;
+use Bio::Chado::Schema;
+use Try::Tiny;
+extends 'CXGN::Metadata::Dbpatch';
+
+
+has '+description' => ( default => <<'' );
+Description of this patch goes here
+
+has '+prereq' => (
+	default => sub {
+        [],
+    },
+
+  );
+
+sub patch {
+    my $self=shift;
+
+    print STDOUT "Executing the patch:\n " .   $self->name . ".\n\nDescription:\n  ".  $self->description . ".\n\nExecuted by:\n " .  $self->username . " .";
+
+    print STDOUT "\nChecking if this db_patch was executed before or if previous db_patches have been executed.\n";
+
+    print STDOUT "\nExecuting the SQL commands.\n";
+
+    my $schema = Bio::Chado::Schema->connect( sub { $self->dbh->clone } );
+
+    my $row = $schema->resultset("Cv::Cv")->find( { name => 'geolocations_property' } );
+
+    if (defined $row) {
+        $row->name('geolocation_property');
```

Comment:
some databases already have a cv called geolocation_property along with geolocations_property (e.g. cassavabase) so the patch will fail on them. in cassavabase there is one cvterm under the geolocation_property

Reply:
uhoh, I do see that now. Any thoughts on the best solution? I suppose I could modify the patch so that it checks for both possible cv names and merges them into a single 'geolocation_property' cv

Reply:
@bellerbrock if the patch could merge them into geolocation_property that would be great. as far as i know there is just that "geolocation address" cvterm that is a problem right now

Reply:
cool. commit 08d0d383859c712f27914fd01847435be6f95c23  updates the patch to handle a merge if necessary. anything else to address before merging?